netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -165,14 +166,20 @@ public ClientStream newStream(MethodDescriptor<?, ?> method, Metadata headers) {
   public Runnable start(Listener transportListener) {
     lifecycleManager = new ClientTransportLifecycleManager(
```

Comment:
oops, I see `lifecycleManager` is also used in `statusFromFailedFuture`. But why did you add `lifecycleManager` as an extra argument of `newHandler`?